### PR TITLE
thunderbird@daily: update various

### PR DESCRIPTION
**Pull request by AlternateRT**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

---
Updates the `homepage` url to be something more up to date, the `name` so that it is more accurate (Thunderbird Daily isn't called Earlybird any more), and makes the `zap` stanza match the one from #187833.

### CODE REVIEW

File: Casks/t/thunderbird@daily.rb [hunk 1]
```diff
@@ -168,21 +168,21 @@
...
   app "Thunderbird Daily.app"
 
   zap trash: [
-    "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/org.mozilla.thunderbird.sfl*",
+    "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/org.mozilla.thunderbird*.sfl*",
     "~/Library/Caches/Thunderbird",
-    "~/Library/Preferences/org.mozilla.thunderbird daily.plist",
-    "~/Library/Saved Application State/org.mozilla.thunderbird.savedState",
+    "~/Library/Preferences/org.mozilla.thunderbird*.plist",
+    "~/Library/Saved Application State/org.mozilla.thunderbird-daily.savedState",
```
Comment (khipp) [inline Casks/t/thunderbird@daily.rb:185]:
```suggestion
    "~/Library/Saved Application State/org.mozilla.thunderbird*.savedState",
```

I think we should not make a distinction here (other PRs as well), at the very least for consistency with the other entries.

Comment (AlternateRT) [reply Casks/t/thunderbird@daily.rb:185]: Sure thing. I'll get to it on the other PRs as well.

**Review by khipp — APPROVED** [review]
@AlternateRT Thank you for doing such a wonderful job getting those casks in line! There has also been an [issue about adding `thunderbird@esr`](https://github.com/Homebrew/homebrew-cask/issues/187251#issuecomment-2391476837), if you’d like to take a stab at it.

**Comment by AlternateRT** [comment]
No problem! I'd be happy to work on adding `thunderbird@esr` as well.

**Pull request merged**